File: app/src/main/java/io/github/thang86/weathers/base/BasePresenter.java
package io.github.thang86.weathers.base;

/**
 * ThangTX2
 */
public class BasePresenter<T extends BaseView> implements MvpPresenter<T> {
    private T mView;

    @Override
    public void attachView(T view) {
        this.mView = view;
    }

    @Override
    public void detachView() {
        this.mView = null;

    }

    /**
     * Method geter get view
     *
     * @return
     */
    public T getMView() {
        return this.mView;
    }


    public boolean isViewAttached() {
        return this.mView != null;
    }

    /**
     * Method check view attached
     *
     * @return
     */
    public void checkViewAttached() {
        if (!isViewAttached()) throw new ViewNotAttachedException();
    }
}

File: app/src/main/java/io/github/thang86/weathers/core/BasePresenter.kt
package io.github.thang86.weathers.core

open class BasePresenter<T : BaseView> : MvpPresenter<T> {

    private var mView: T? = null

    override fun attachView(view: T) {
        this.mView = view
    }

    override fun detachView() {
        mView = null
    }

    open fun getMView(): T? {
        return this.mView
    }

    fun isViewAttached(): Boolean {
        return this.mView != null
    }

    fun checkViewAttached() {
        if (!isViewAttached()) throw ViewNotAttachedException()
    }
}